src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -1967,6 +1967,46 @@ internal MethodSymbol EnsureInlineArrayAsReadOnlySpanExists(SyntaxNode syntaxNod
             return (MethodSymbol)privateImplClass.GetMethod(PrivateImplementationDetails.SynthesizedInlineArrayAsReadOnlySpanName)!.GetInternalSymbol()!;
         }
 
+        internal MethodSymbol EnsureInlineArrayElementRefExists(SyntaxNode syntaxNode, NamedTypeSymbol intType, DiagnosticBag diagnostics)
```

Comment:
>EnsureInlineArrayElementRefExists [](http://example.com/codeflow?start=30&length=33)

It looks like these new methods and the methods above have the same boilerplate for the GetMethod(), TryAddSynthesizedMethod(), GetMethod() calls. Consider extracting a helper method that takes a method name and a delegate to create the method.

Reply:
> It looks like these new methods and the methods above have the same boilerplate for the GetMethod(), TryAddSynthesizedMethod(), GetMethod() calls. Consider extracting a helper method that takes a method name and a delegate to create the method.

Since this refactoring going to include methods that are not related to the feature. I prefer to implement this refactoring in a separate PR, if there are no objections.